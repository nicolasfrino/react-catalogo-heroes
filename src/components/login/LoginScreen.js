import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ( { history }) => {

    const { dispatch } = useContext(AuthContext);

    const handleLogin = () => {
        //console.log('history', history);
        //history.push('/'); //uno deja navegar para atras y el otro como que pirde la historia
        //history.replace('/');

        const lastPath = localStorage.getItem('lastPath') || '/';


        dispatch ({
            type: types.login,
            payload: {
                name: 'Nicolas'
            }
        })

        history.replace(lastPath);


    };

    return (
        <div className='container mt-5'>
            <h1>LoginScreen</h1>
            <hr />

            <button className='btn-primary'
                    onClick= { handleLogin }
            >
                Login
            </button>
        </div>
    )
}
