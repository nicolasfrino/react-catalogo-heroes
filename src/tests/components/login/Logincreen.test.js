import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

describe('LoginScreen', () => {
    
    const contextValues= {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
        
    }

    const history = {
        replace: jest.fn()
    }

    const wrapper = mount (
        <AuthContext.Provider value={contextValues}>
            <LoginScreen history={history}/>
        </AuthContext.Provider>
    )

    test('debe mostarse correctamente', () => {
        
        expect (wrapper).toMatchSnapshot();

    })
    
    test('debe realizar dispatch y navegacion', () => {
        
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect( contextValues.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Nicolas'
            }
        })

        expect (history.replace).toHaveBeenCalled();
        expect (history.replace).toHaveBeenCalledWith('/');

        localStorage.setItem('lastPath','/dc');
        handleClick();

        expect (history.replace).toHaveBeenCalledWith('/dc');



    })

})
