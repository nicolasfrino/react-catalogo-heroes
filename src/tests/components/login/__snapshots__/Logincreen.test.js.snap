// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen debe mostarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
